### PR TITLE
phoenix-pilot: new mathematical model (attitude + position, 2/3)

### PR DESCRIPTION
**Pull request by niewim19**
## Description

Ekf new mathematical model is expanded on position estimation (currently only vertical through barometer):
 - addition of velocity and position estimation
 - accelrometer measurements usage in ekf prediction step in integration equations
 - introduces low pass filter on gyro data
 - `devekf` silent mode (quietly run ekf without any logging) enable better development of ekf

**Note:** this branch is rebased on top of https://github.com/phoenix-pilot/phoenix-pilot-core/pull/189. First six commits belong to linked branch.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: ekf/filters.c [hunk 1]
```diff
@@ -100,6 +100,64 @@ static const float fltr_accWindow[FLTR_ACCEL_LEN] = {
...
+    0.086741445655463434,
+    0.091476727498003868,
+    0.094950333264530992,
+    0.097071580192293394,
+    0.097784901530386506,
+    0.097071580192293394,
+    0.094950333264530992,
+    0.091476727498003868,
+    0.086741445655463434,
+    0.080867462624515052
```
Comment (kemonats) [inline ekf/filters.c:116]:
```suggestion
	0.080867462624515052,
	0.086741445655463434,
	0.091476727498003868,
	0.094950333264530992,
	0.097071580192293394,
	0.097784901530386506,
	0.097071580192293394,
	0.094950333264530992,
	0.091476727498003868,
	0.086741445655463434,
	0.080867462624515052
```
tabs instead spaces

File: ekf/filters.c [hunk 2]
```diff
@@ -135,10 +193,64 @@ static void fltr_windowVec(vec_t *raw, vec_t *buf, int *bufPos, const float *win
...
+
+
+void fltr_gyroLpf(vec_t *raw)
+{
+	static vec_t buf[FLTR_GYRO_LEN] = { 0 };
+	static int bufPos = 0;
+
+	fltr_windowVec(raw, buf, &bufPos, fltr_gyroWindow, FLTR_GYRO_LEN);
+}
+
```
Comment (kemonats) [inline ekf/filters.c:256]:
```suggestion
```
remove empty line at end of file

File: ekf/kalman_update_baro.c [hunk 3]
```diff
@@ -24,22 +24,21 @@
...
 #include <matrix.h>
 
 
-enum baroDimension { value = 0, dtime = 1 };
-
-
 /* Rerurns pointer to passed Z matrix filled with newest measurements vector */
 static matrix_t *getMeasurement(matrix_t *Z, matrix_t *state, matrix_t *R, time_t timeStep)
 {
 	static uint64_t lastTstamp = 0;
```
Comment (kemonats) [inline ekf/kalman_update_baro.c:37]: maybe of type `time_t`

File: ekf/kalman_update_baro.c [hunk 4]
```diff
@@ -24,22 +24,21 @@
...
 	static float lastAlt = 0;
+	static float lpfAltChange = 0;
 
-	float pressure, temp;
+	float pressure, temp, alt;
 	uint64_t currTstamp;
```
Comment (kemonats) [inline ekf/kalman_update_baro.c:42]: maybe of type `time_t`

File: utils/devekf.c [hunk 5]
```diff
@@ -68,6 +68,9 @@ int main(int argc, char **argv)
 	if (atoi(argv[1]) == 2) {
 		mode = printAcc;
 	}
+	if (atoi(argv[1]) == 3) {
+		mode = silent;
```
Comment (kemonats) [inline utils/devekf.c:72]: instaed of using `atoi()`
```c
switch (argv[1][0]) {
	case '1':
		mode = prntAtt;
		break;
	case '2':
		mode = printAcc;
		break;
	case '3':
		mode = silent;
		break;
	default:
		break;
}

File: ekf/filters.c (hunk 1 above)
Comment (niewim19) [reply ekf/filters.c]: Ugly copy paste from me. Thanks for catching!

File: ekf/kalman_update_baro.c (hunk 3 above)
Comment (niewim19) [reply ekf/kalman_update_baro.c]: Implemented, thanks!

File: ekf/kalman_update_baro.c (hunk 4 above)
Comment (niewim19) [reply ekf/kalman_update_baro.c]: Implemented, thanks!

File: utils/devekf.c (hunk 5 above)
Comment (niewim19) [reply utils/devekf.c]: Implemented, thanks!

**Comment by niewim19** [comment]
This PR is being closed and commits introduced in it are merged in a cumulative PR:
 - https://github.com/phoenix-pilot/phoenix-pilot-core/pull/199

**Pull request closed without merging**